### PR TITLE
Remove option for checkout type in platform

### PR DESCRIPTION
**Pull request by phieronymus**
Ok, that should hopefully be the last one. Sorry for the mistakes 🙂

### CODE REVIEW

**Comment by phieronymus** [comment]
Adding Kingston to the list of reviewers as the optional thing is a bit confusing to me when reading the code. I apologize for the inconvenience.

**Comment by phieronymus** [comment]
Adding Kingston to the list of reviewers as the optional thing is a bit confusing to me when reading the code. I apologize for the inconvenience.

File: src/resources/common/Configuration.ts [hunk 1]
```diff
@@ -96,8 +96,8 @@ export interface DescriptorProvidedConfiguration {
     phoneNumber: string;
 }
 export type CheckoutConfiguration = {
-    ecEmail: { enabled: boolean };
-    ecProduct: { enabled: boolean };
+    ecEmail?: { enabled?: boolean };
```
Comment (perrin4869) [inline src/resources/common/Configuration.ts:99]: `ecEmail? -> ecEmail`

File: src/resources/common/Configuration.ts [hunk 2]
```diff
@@ -96,8 +96,8 @@ export interface DescriptorProvidedConfiguration {
...
+    ecProduct?: { enabled?: boolean };
```
Comment (perrin4869) [inline src/resources/common/Configuration.ts:100]: `ecProduct? -> ecProduct`

**Review by perrin4869 — CHANGES_REQUESTED** [review]
Checked yesterday, the `enabled` part may be omitted, but `ecEmail: {}` and `ecProducts: {}` will always be output.
Oh, and it's `ecProducts` instead of `ecProduct` ><;;

File: test/fixtures/common/configuration.ts [hunk 3]
```diff
@@ -80,6 +80,10 @@ export const generateFixture = (): ConfigurationItem => ({
         suspendOnCancel: true,
         failedChargesToCancel: 10,
     },
+    checkoutConfiguration: {
+        ecEmail: { enabled: true },
+        ecProduct: { enabled: true },
```
Comment (perrin4869) [inline test/fixtures/common/configuration.ts:85]: `ecProducts`

File: test/fixtures/platforms.ts [hunk 4]
```diff
@@ -35,6 +35,10 @@ export const generateFixture = (): PlatformConfigurationItem => ({
                 amount: 1000,
                 currency: "JPY",
             },
+            checkoutConfiguration: {
+                ecEmail: { enabled: true },
+                ecProduct: { enabled: true },
```
Comment (perrin4869) [inline test/fixtures/platforms.ts:40]: `ecProducts`

File: src/resources/common/Platform.ts [hunk 5]
```diff
@@ -1,13 +1,17 @@
...
     InstallmentsConfiguration,
     SubscriptionsConfiguration,
     UserTransactionsConfiguration,
 } from "./Configuration";
 import { TransferScheduleItem } from "./TransferSchedule";
 import { AmountWithCurrency, InvoiceChargeFee, WithCreatedOn } from "./types";
 
+type CheckoutConfiguration = {
+    ecEmail: { enabled: boolean };
+    ecProduct: { enabled: boolean };
```
Comment (perrin4869) [inline src/resources/common/Platform.ts:12]: `ecProducts`

**Review by perrin4869 — CHANGES_REQUESTED** [review]

**Pull request merged**